src/com/geekhub/hw2/cars/SolarCar.java

```diff
@@ -0,0 +1,24 @@
+package com.geekhub.hw2.cars;
+
+import com.geekhub.hw2.parts.interfaces.EnergySource;
+import com.geekhub.hw2.parts.interfaces.Engine;
+import com.geekhub.hw2.parts.interfaces.Gear;
+import com.geekhub.hw2.parts.interfaces.RunMechanism;
+
+public class SolarCar extends Car {
+
+    public SolarCar(EnergySource energySource, Engine engine, Gear gear, RunMechanism runMechanism) {
```

Comment:
I think it's mistake in hometask itself. Should be abstraction Car and two children GasCar and SolarCar. Because what means inheritance. It means that child type should also be parent type. Forexample Cat type is also Animal type. So it's incorrect that SolarCar type is also GasCar type.